Makefile

```diff
@@ -1,321 +1,259 @@
-SCORECARD_TEST_KUTTL_IMAGE ?= $(SCORECARD_TEST_KUTTL_BASE_IMAGE)
-
-ANSIBLE_ARCHES:="amd64" "ppc64le" "arm64" "s390x"
-HELM_ARCHES:="amd64" "ppc64le" "arm64" "s390x"
-OPERATOR_SDK_ARCHES:="amd64" "ppc64le" "arm64" "s390x"
-SCORECARD_TEST_ARCHES:="amd64" "ppc64le" "arm64" "s390x"
-SCORECARD_TEST_KUTTL_ARCHES:="amd64" "ppc64le" "arm64"
-# the custom scorecard test image is a scorecard example only
-CUSTOM_SCORECARD_TESTS_ARCHES:="amd64" "ppc64le" "arm64"
-
-export CGO_ENABLED:=0
-.DEFAULT_GOAL:=help
-
-.PHONY: help
-help: ## Show this help screen
-	@echo 'Usage: make <OPTIONS> ... <TARGETS>'
-	@echo ''
-	@echo 'Available targets are:'
-	@echo ''
-	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
-
-##############################
-# Development                #
-##############################
+export GO111MODULE = on
+export CGO_ENABLED = 0
+export PATH := $(BUILD_DIR):$(PWD)/tools/bin:$(PATH)
 
 ##@ Development
 
-.PHONY: all install
-
-all: format test build/operator-sdk ## Test and Build the Operator SDK
-
-install: ## Install the binaries
-	$(Q)$(GOARGS) go install $(GO_BUILD_ARGS) ./cmd/operator-sdk ./cmd/ansible-operator ./cmd/helm-operator
-
-
-# Code management.
-.PHONY: format tidy clean cli-doc lint
-
-format: ## Format the source code
-	$(Q)go fmt $(PKGS)
-
-tidy: ## Update dependencies
-	$(Q)go mod tidy -v
-
-clean: ## Clean up the build artifacts
-	$(Q)rm -rf build
-
-lint: ## Install and run golangci-lint checks
-ifneq (${GOLANGCI_LINT_VER}, "$(shell ./bin/golangci-lint --version 2>/dev/null | cut -b 27-32)")
-	@echo "golangci-lint missing or not version '${GOLANGCI_LINT_VER}', downloading..."
-	curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/v${GOLANGCI_LINT_VER}/install.sh" | sh -s -- -b ./bin "v${GOLANGCI_LINT_VER}"
-endif
-	./bin/golangci-lint --timeout 5m run
-
-setup-k8s:
-	hack/ci/setup-k8s.sh ${K8S_VERSION}
-
-##############################
-# Generate Artifacts         #
-##############################
-
-##@ Generate
-
-.PHONY: generate cli-doc changelog samples
-
-generate: cli-doc samples bindata ## Run all non-release generate targets
-
-cli-doc: ## Generate CLI documentation
+.PHONY: generate
+generate: build # Generate CLI docs and samples
 	go run ./hack/generate/cli-doc/gen-cli-doc.go
-
-samples: ## Generate samples
 	go run ./hack/generate/samples/generate_all.go
 
-changelog: ## Generate CHANGELOG.md and migration guide updates
-	./hack/generate/changelog/gen-changelog.sh
-
-bindata:
+.PHONY: bindata
+OLM_VERSION=0.15.1
+bindata: ## Update project bindata
 	./hack/generate/olm_bindata.sh $(OLM_VERSION)
 
-##############################
-# Release                    #
-##############################
-
-##@ Release
+.PHONY: fix
+fix: ## Fixup files in the repo.
+	go mod tidy
+	go fmt ./...
```

Comment:
WDYT about to add here the lint-fix so?

Reply:
I'm hesitant to put that in the critical path of PRs. What if we don't agree with one of the fixes that it makes? I've never personally used the `--fix` flag or the `lint-fix` target, so I don't have first-hand experience, but it always struck me as a convenience wrapper to run if the linter failed to quickly resolve whatever it's complaining about. And then contributors would have a chance to manually review the changes before committing.

Reply:
the --fix usually fix spaces and go fmt issues. It does not fix all scenarios. if we add the lint check gofmt in https://github.com/operator-framework/operator-sdk/blob/master/.golangci.yml#L2 which I think should have, then we can replace the go fmt ./... here by the lint fix.